hazelcast/src/main/java/com/hazelcast/config/EndpointConfig.java

```diff
@@ -274,10 +278,35 @@ public EndpointConfig setSSLConfig(SSLConfig sslConfig) {
         return this;
     }
 
+    /**
```

Comment:
The nullability should be documented. @Nonnull has no meaning with respect to javadoc.

Reply:
You mean the return values? Because otherwise all parameters have `@throws`. For example:
```
    /**
     * Sets the Alto socket config.
     *
     * @param altoSocketConfig Alto socket config to set
     * @return this endpoint config
     * @throws NullPointerException if altoSocketConfig is null
     * @see com.hazelcast.config.alto.AltoConfig
     * @since 5.3
     */
    @Beta
    @Nonnull
    public EndpointConfig setAltoSocketConfig(@Nonnull AltoSocketConfig altoSocketConfig) {
        this.altoSocketConfig = checkNotNull(altoSocketConfig);
        return this;
    }
```
see
```
     * @throws NullPointerException if altoSocketConfig is null
```

Reply:
A user should know if the returned EndpointConfig will be null or if you need to add a guard.

Reply:
https://github.com/hazelcast/hazelcast/pull/23319/commits/d9d7f46f1cc90fcc5e14f07aa8c76e19d66b4e46